lib/puppet/pops/types/p_sensitive_type.rb

```diff
@@ -0,0 +1,69 @@
+module Puppet::Pops
+module Types
+
+# A Puppet Language type that wraps sensitive information. The sensitive type is parameterized by
+# the wrapped value type.
+#
+#
+# @api public
+class PSensitiveType < Puppet::Pops::Types::PScalarType
+
+  class Sensitive
+    def initialize(value)
+      @value = value
+    end
+
+    def unwrap
+      @value
+    end
+
+    def to_s
+      "Sensitive"
+    end
+
+    def inspect
+      "#<Sensitive>"
+    end
+  end
+
+  attr_reader :element_type
+
+  def initialize(element_type = Object)
+    @element_type = element_type
+  end
+
+  def instance?(o, guard = nil)
+    o.is_a?(Sensitive)
+  end
+
+  def eql?(o)
+    self.class == o.class
```

Comment:
for two instances of Sensitive to be equal, their @element_type attributes must also be equal - that is; Sensitive[Integer] == Sensitive[Integer], and Sensitive[Numeric] != Sensitive[Integer]